.github/workflows/test_buildx_and_publish.yml

```diff
@@ -40,14 +40,53 @@ jobs:
           curl --fail http://127.0.0.1:8000/test.php
           curl --fail http://127.0.0.1:8000/check-ini.php
 
+          # Test the default DocumentRoot when it contains a public directory.
+          docker run --name test1 -d -p 8002:80 \
+              -v $PWD/tests/fixtures/documentroot:/var/www/html \
+              moodle-php-apache
+          sleep 1 # Wait for the server to start
+          # Test the DocumentRoot when it contains a public directory.
+          curl --fail http://127.0.0.1:8002/test.php
+
+          # Test the DocumentRoot when it does not contain a public directory.
+          docker run --name test2 -d -p 8003:80 \
+              -v $PWD/tests/fixtures/documentroot/public:/var/www/html \
+              moodle-php-apache
+          sleep 1 # Wait for the server to start
+          curl --fail http://127.0.0.1:8003/test.php
+
+          # Test the DocumentRoot when it is set to a different directory.
+          docker run --name test3 -d -p 8004:80 \
+              -v $PWD/tests/fixtures/documentroot:/srv/moodle \
+              -e APACHE_DOCUMENT_ROOT=/srv/moodle \
+              moodle-php-apache
+          sleep 1 # Wait for the server to start
+          curl --fail http://127.0.0.1:8004/notpublic.php
+
+          # Test the DocumentRoot is not guessed when it has been overriden.
+          docker run --name test4 -d -p 8005:80 \
+              -v $PWD/tests/fixtures/documentroot/public:/srv/moodle \
+              -e APACHE_DOCUMENT_ROOT=/srv/moodle \
+              moodle-php-apache
+          sleep 1 # Wait for the server to start
+          curl --fail http://127.0.0.1:8005/test.php
+
       - name: Display container logs on failure
         if: failure()
         run: |
           docker logs test0
+          docker logs test1
+          docker logs test2
+          docker logs test3
+          docker logs test4
 
       - name: Cleanup docker images
```

Comment:
Minor detail - this is cleanup of docker containers (is it really needed since job is ephemeral?).